defaultNatsService.go

```diff
@@ -0,0 +1,346 @@
+	if n.nc == nil {
+		return fmt.Errorf("can't subscribe. no connection to nats server")
+	}
+	if n.checkSubscriptionExist(subj, queue) {
+		return fmt.Errorf("subscription for subj:%v and queue:%v already exists", subj, queue)
+	}
+	subscription, err := n.nc.QueueSubscribe(subj, queue, func(msg *nats.Msg) {
+		resp := handler.Exec(msg.Data)
+		err := msg.Respond(resp)
+		n.logger.Error(context.Background(), "can't send response on queue subscribe subj: %v, queue: %v. error: %v", subj, queue, err)
+	})
+	if err != nil {
+		return fmt.Errorf("can't queue subscribe subj: %v, queue: %v. error: %v", subj, queue, err)
+	}
+	n.addSubscription(subj, queue, subscription)
+	return nil
+}
+
+func (n defaultNatsService) Subscribe(subj string, handler NatsHandler) error {
+	return n.QueueSubscribe(subj, "", handler)
+}
+
+func (n *defaultNatsService) SubscribeExternal(subj string, handler NatsHandlerPlainData) error {
+	return n.QueueSubscribeExternal(subj, "", handler)
+}
+
+func (n defaultNatsService) respond(subj, queue string, msg *nats.Msg, b []byte) {
+	err := n.nc.Publish(msg.Reply, b)
+	if err != nil {
+		n.logger.Error(context.Background(), "can't send response on queue subscribe subj: %v, queue: %v. error: %v", subj, queue, err)
+	}
+}
+
+func (n *defaultNatsService) checkSubscriptionExist(subj, queue string) bool {
+	_, ok := n.subscriptions[subj+queue]
+	return ok
+}
+
+func (n *defaultNatsService) addSubscription(subj, queue string, subscription *nats.Subscription) {
+	n.subscriptions[subj+queue] = subscription
+}
+
+func handleInternalResponse(body []byte, target interface{}) (srvReply ServiceReply) {
+	var srvError Response
+	srvReply = NewNil()
+	if err := json.Unmarshal(body, &srvError); err != nil {
+		return NewInternalServiceError(err).WithLogMessage(fmt.Sprintf("cannot read response from")).WithLogValues(ValuesMap{"rawResponse": string(body)})
+	}
+	if srvError.Status != status.SuccessStatus {
+		resType := status.GetTypeByStatus(srvError.GetStatus())
+		msgValues := srvError.GetMessageValues()
+		srvReply = NewServiceError(&resType, fmt.Errorf(string(srvError.GetStatus())+". "+srvError.GetMessageId()), srvError.GetMessageId(), 1)
+		if msgValues != nil {
+			srvReply = srvReply.WithReplyValues(*msgValues)
+		}
+		return srvReply
+	}
+	if srvError.Message != nil {
+		msgValues := srvError.GetMessageValues()
+		srvReply = NewMessage(srvError.GetMessageId())
+		if msgValues != nil {
+			srvReply = srvReply.WithReplyValues(*msgValues)
+		}
+	}
+	if srvError.Data != nil {
+		if dataJson, err := json.Marshal(srvError.Data); err != nil {
+			return NewInternalServiceError(fmt.Errorf("can't unmarshal serviceReply response. " + err.Error()))
+		} else {
+			body = dataJson
+		}
+	} else {
+		body = nil
+	}
+	if err := json.Unmarshal(body, &target); err != nil {
+		return NewInternalServiceError(fmt.Errorf("can't unmarshal response data. " + err.Error()))
```

Comment:
perhaps all these marshal failures should also use same func that will also get the DATA
then if log level > info we will log it